modules/terraform/output.go

```diff
@@ -131,16 +135,71 @@ func OutputMapE(t *testing.T, options *Options, key string) (map[string]string,
 	return resultMap, nil
 }
 
-// OutputForKeysE calls terraform output for the given key list and returns values as a map.
+// OutputForKeys calls terraform output for the given key list and returns values as a map.
 // If keys not found in the output, fails the test
 func OutputForKeys(t *testing.T, options *Options, keys []string) map[string]interface{} {
 	out, err := OutputForKeysE(t, options, keys)
 	if err != nil {
 		t.Fatal(err)
 	}
+
 	return out
 }
 
+// OutputForKeysTgE calls terragrunt output-all for the given key list and returns values as a map.
+// If keys not found in the output, fails the test
+func OutputForKeysTgE(t *testing.T, options *Options, keys []string) (map[string]interface{}, error) {
+	out, err := RunTerraformCommandE(t, options, "output-all", "-no-color", "-json")
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse out json from terragrunt output
+	outList := strings.Split(out, "\n")
+	for index, line := range outList {
+		if line == "{" {
+			outList = outList[index:]
+			break
+		}
+	}
+
+	// Parse out strings in json block that are not json.
+	// Due to how readStdoutAndStderr threads, it's possible
+	// to get terragrunt output in the json
+	for index, line := range outList {
+		if strings.Contains(line, "[terragrunt]") {
+			outList = append(outList[:index], outList[index+1:]...)
+		}
+	}
```

Comment:
I think it would be better to construct a different outlist, filtering out terragrunt first. E.g:

```
// filter out all the terragrunt logs
var outList := []string{}
for _, line := strings.Split(out, "\n") {
    if !strings.HasPrefix(line, "[terragrunt]") {
        outList = append(outList, line)
    }
}
```

Reply:
The benefit of this approach is that we aren't modifying the original list as we are going through it, which is awkward can lead to subtle bugs.